bodhi/tests/server/test_utils.py

```diff
@@ -70,3 +72,50 @@ def test_sorted_builds(self):
         b1, b2 = sorted_builds([new, old])
         assert b1 == new, b1
         assert b2 == old, b2
+
+
+class TestCMDFunctions(unittest.TestCase):
+    @mock.patch('subprocess.Popen')
+    def test_stderr(self, mock_popen):
+        """
+        Ensures proper behavior when there is err output and the exit code isn't 0.
+        See https://github.com/fedora-infra/bodhi/issues/1412
+        """
+        mock_popen.return_value = mock.Mock()
+        mock_popen_obj = mock_popen.return_value
+        mock_popen_obj.communicate.return_value = ('output', 'error')
+        mock_popen_obj.returncode = 1
+        with mock.patch('bodhi.server.log.error') as mock_error:
+            cmd('/bin/echo', '"Hello World"')
+            self.assertTrue(mock_popen.called)
```

Comment:
Let's make a stronger assertion than just that popen was called - you can use something like this:

```
mock_popen.assert_called_once_with(['bin/echo', '"Hello World"'], cwd=None, stdout=subprocess.PIPE,
                                   stderr=subprocess.PIPE)
```

This would be good to do in the other tests as well.